publish TA #144650 (#19564)
cockroachdb/docs: src/current/advisories/a144650.md

This issue only affects the following operations: `IMPORT`, MOLT Fetch (without Verify), Physical Cluster Replication, `CREATE TABLE ... AS` (CTAS), and materialized view refreshes.

Users on affected versions can determine whether they may have encountered the problem by running the script [`detect_144650.sh`](detect_144650.sh) to detect any bulk ingestion jobs that improperly succeeded. If the script detects a potential issue, users should [contact support](https://support.cockroachlabs.com/) and work through the appropriate [manual validation](#manual-validation) steps.

For details, refer to [Detection via logs](#detection-via-logs) and [Manual validation](#manual-validation).

### Detection via logs

[`detect_144650.sh`](detect_144650.sh) requires the following dependencies:

- [ripgrep](https://github.com/BurntSushi/ripgrep#installation)
- [DuckDB](https://duckdb.org/docs/installation)

Run `detect_144650.sh` from within a decompressed [`debug.zip`]({% link v25.1/cockroach-debug-zip.md %}) directory or by appending the path to the directory. For example:

{% include_cached copy-clipboard.html %}
~~~ shell
/path/to/script/detect_144650.sh /path/of/unzipped/debug
~~~

> Do not run the script on a production system. Instead, move the `debug.zip` file elsewhere for analysis and run the script there.

Reviewer: ```suggestion
Do not run the script on a production system. Instead, move the `debug.zip` file to a separate system for analysis and run the script there.
```

- If you ran MOLT Verify after completing your MOLT Fetch run, and Verify did not find mismatches, then MOLT Fetch was unaffected by this issue.

- If you did not run Verify after Fetch, analyze the exported files that exist in your configured [Fetch data path]({% link molt/molt-fetch.md %}#data-path) to determine the expected number of rows. Then follow steps 1-3 in the [`IMPORT`](#import) section.

#### Physical Cluster Replication

After upgrading both the primary and standby clusters to a [version with the fix](#mitigation), run a [point-in-time fingerprint on both clusters]({% link v25.1/physical-cluster-replication-monitoring.md %}#data-verification) to verify that the data on the primary and standby clusters are the same at a certain [replicated time]({% link v25.1/physical-cluster-replication-technical-overview.md %}#failover-and-promotion-process).

> If the fingerprints do not have the same value, then [upgrade to a version with the fix](#mitigation), cancel your PCR stream by [dropping the replicating virtual cluster]({% link v25.1/drop-virtual-cluster.md %}), and [restart the PCR stream]({% link v25.1/set-up-physical-cluster-replication.md %}). Cockroach Labs encourages users to [contact support](https://support.cockroachlabs.com/) to work through these steps if a fingerprint mismatch is found.

Reviewer: ```suggestion
If the fingerprints do not have the same value, then [upgrade to a version with the fix](#mitigation), cancel your PCR stream by [dropping the replicating virtual cluster]({% link v25.1/drop-virtual-cluster.md %}), and [restart the PCR stream]({% link v25.1/set-up-physical-cluster-replication.md %}). Cockroach Labs encourages users to [contact support](https://support.cockroachlabs.com/) to assist with these steps if a fingerprint mismatch is found.
```

#### `CREATE TABLE ... AS` (CTAS)

[`CREATE TABLE ... AS`]({% link v25.1/create-table-as.md %}) (CTAS) operations can be validated similarly to [`IMPORT`](#import) jobs. The number of rows in the created table indicates whether the CTAS operation was successful.

> To check whether a CTAS operation was affected, compare the number of rows in the created table with the output of the CTAS query. If this comparison is not possible due to the time elapsed since the CTAS statement was run, then either re-run the CTAS query with [`AS OF SYSTEM TIME`]({% link v25.1/as-of-system-time.md %}) or use [`RESTORE ... AS OF SYSTEM TIME`]({% link v25.1/restore.md %}#restore-with-as-of-system-time) to reconstruct the table as it was immediately after the CTAS operation finished.

Reviewer: ```suggestion
To check whether a CTAS operation was affected, compare the number of rows in the created table with the output of the CTAS query. If you cannot perform this comparison due to the time elapsed since the CTAS operation, then either re-run the CTAS query with [`AS OF SYSTEM TIME`]({% link v25.1/as-of-system-time.md %}) or use [`RESTORE ... AS OF SYSTEM TIME`]({% link v25.1/restore.md %}#restore-with-as-of-system-time) to reconstruct the table as it was immediately after the CTAS operation finished.
```
